### ISSUE TITLE
Issue deploying model to Power BI Premium

### ISSUE THREAD

**Issue by fhmessas** [TabularEditor/TabularEditor#1062]
Hi, I'm using a SPN with all permissions to deploy a powerbi model, however, I'm facing an unhandled exception. I've done all checks for the SPN, deployed manually but nothing seems to work. 
I'm using Azure Devops Pipeline with Powershell script.

As additional information, when I remove the instruction **"Initial Catalog=B2B;"** I get the error "Workspace not found".

Please check the logs below.

`Version      : 2.212.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
##[debug]VstsTaskSdk 0.9.0 commit 6c48b16164b9a1c9548776ad2062dad5cd543352
##[debug]Entering D:\a\_tasks\CmdLine_d9bafed4-0b18-4f58-968d-86655b4d2ce9\2.212.0\cmdline.ps1.
##[debug]Loading resource strings from: D:\a\_tasks\CmdLine_d9bafed4-0b18-4f58-968d-86655b4d2ce9\2.212.0\task.json
##[debug]Loaded 6 strings.
##[debug]SYSTEM_CULTURE: 'en-US'
##[debug]Loading resource strings from: D:\a\_tasks\CmdLine_d9bafed4-0b18-4f58-968d-86655b4d2ce9\2.212.0\Strings\resources.resjson\en-US\resources.resjson
##[debug]Loaded 6 strings.
##[debug]INPUT_FAILONSTDERR: 'false'
##[debug] Converted to bool: False
##[debug]INPUT_SCRIPT: '@TabularEditor.exe "D:\a\1/_pbi/Model.bim" -S "D:\a\1/_pbi_utils/ChangeConnectionString.cs" -D "Data Source=powerbi://api.powerbi.com/v1.0/myorg/I&E DE IAM Identity Application and Access Analytics Assessment (B2B) (DEV);Initial Catalog=B2B;User ID=app:<appid>@<tenantId>;Password=***;Persist Security Info=True;Impersonation Level=Impersonate;User ID=app:<appid>@<tenantId>;Password=***;Persist Security Info=True;Impersonation Level=Impersonate" "B2B" -O -C -P -E -R -M'
##[debug]INPUT_WORKINGDIRECTORY: 'D:\a\1\s'
##[debug]Asserting container path exists: 'D:\a\1\s'
Generating script.
Script contents: shell
@TabularEditor.exe "D:\a\1/_pbi/Model.bim" -S "D:\a\1/_pbi_utils/ChangeConnectionString.cs" -D "Data Source=powerbi://api.powerbi.com/v1.0/myorg/I&E DE IAM Identity Application and Access Analytics Assessment (B2B) (DEV);Initial Catalog=B2B;User ID=app:<appid>@<tenantId>;Password=***;Persist Security Info=True;Impersonation Level=Impersonate;User ID=app:<appid>@<tenantId>;Password=***;Persist Security Info=True;Impersonation Level=Impersonate" "B2B" -O -C -P -E -R -M
##[debug]AGENT_VERSION: '2.213.2'
##[debug]AGENT_TEMPDIRECTORY: 'D:\a\_temp'
##[debug]Asserting container path exists: 'D:\a\_temp'
##[debug]Asserting leaf path exists: 'C:\Windows\system32\cmd.exe'
========================== Starting Command Output ===========================
##[debug]Entering Invoke-VstsTool.
##[debug] Arguments: '/D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\b4614d1c-9914-4f6c-bee2-1786fb2ce3d3.cmd""'
##[debug] FileName: 'C:\Windows\system32\cmd.exe'
##[debug] WorkingDirectory: 'D:\a\1\s'
"C:\Windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\b4614d1c-9914-4f6c-bee2-1786fb2ce3d3.cmd""

Tabular Editor 2.17.2 (build 2.17.8336.21072)
--------------------------------
Loading model...
Loaded script: D:\a\1/_pbi_utils/ChangeConnectionString.cs
Executing script 0...
Deploying...
**Deployment failed! COM error: Microsoft.MWC.Workload.Client.Library, {"error":{"code":"PowerBIEntityFrameworkOperationException","pbi.error":{"code":"PowerBIEntityFrameworkOperationException","parameters":{},"details":[]}}}.**

Technical Details:
RootActivityId: 2561bc46-3f17-4e59-b0d3-bb99c0afb675
Date (UTC): 12/15/2022 4:53:43 PM

##[debug]Exit code: 1
##[debug]Leaving Invoke-VstsTool.
##[error]Cmd.exe exited with code '1'.
##[debug]Processed: ##vso[task.logissue type=error]Cmd.exe exited with code '1'.
##[debug]Processed: ##vso[task.complete result=Failed]Error detected
##[debug]Leaving D:\a\_tasks\CmdLine_d9bafed4-0b18-4f58-968d-86655b4d2ce9\2.212.0\cmdline.ps1.
Finishing: CmdLine`

**Comment by fhmessas** [comment]
As additional info, I've tried running the command locally and got the same error. 
The SPN has all permission needed in our PowerBI service.

**Comment by victoradmc** [comment]
Having the same issue here, any known fixes?

**Comment by danielfilhocoelho** [comment]
Same problem here.

**Comment by mlonsk** [comment]
Hi @fhmessas 
CC: @victoradmc , @danielfilhocoelho 

We have not been able to reproduce this error and we are able to successfully deploy to a PBI workspace using a service principal. (See bellow for a screenshot)
We have tried both using the command prompt and a powershell script and it works for us in both instances. 

Our conclusion is therefore that this is not a Tabular Editor issue, but more likely a Microsoft issue. We would advise you to open a ticket with Microsoft support citing the RootActivityID in your output, and see if they can identify what the issue is. 
The error message is also not something we have seen before. 

![image](https://user-images.githubusercontent.com/11939114/211810750-76cbd0fa-f757-4061-b80e-daf09b477c69.png)

We did manage to provoke the "Workspace not found" error, but only by disabling the Power BI tennant setting "Allow service principals to use Power BI APIs":

<img width="296" alt="image" src="https://user-images.githubusercontent.com/11939114/211811420-4af37809-6f8e-470a-9004-c1365b0f6dce.png">
Error message when Tennant setting is disabled:
<img width="352" alt="image" src="https://user-images.githubusercontent.com/11939114/211811640-c3865686-d299-46bb-a5a3-e646440a050a.png">

I wish there was more we could do to help
